### PR TITLE
Update everything

### PR DESCRIPTION
**Pull request by frothedoatmilk**
~~This was the latest version I could get to before bumping the bundler version, since [middleman 4.4 started requiring `bundle ~> 2.0`](https://github.com/middleman/middleman/commit/9b90979f27318b9629f601fb0e6d8fe658685bf5).~~ Bumping ruby, bundler, and middleman to their latest versions.

middleman has [deprecated middleman-sprockets in favor of their `external_pipeline` system](https://github.com/middleman/middleman-sprockets/issues/132#issuecomment-538590940), thus the two new calls to that. We don't need an explicit command because these assets are downloaded as part of the gem when we bundle! When the site builds, it drags over some copies of those files into the appropriate directories.

This change also now makes all of Bootstrap's JS tooling available _if we want it._ We're not requiring it in `all.js` atm so it's not actually being loaded on the page for the time being.

For some reason, I still need to manually tell SassC where the Bootstrap files are. Not the Font Awesome ones though! I've poked around for a while trying to figure out what's causing this to no avail. No config edit leads to a SassC compile error.

### CODE REVIEW

**Comment by frothedoatmilk** [comment]
Also, it appears that the new way Middleman handles Sass files involves trimming off the file extension as it goes. So I dropped in a `.css` into its filename. Without it, the stylesheet ends up as `/stylesheets/all` instead of `/stylesheets/all.css`

**Comment by sfan5** [comment]
I've been messing around for 20 minutes and getting a working ruby 2.7 environment set up has turned out to be harder than expected. You'll have to wait a bit more for me to review this.

**Comment by kasper93** [comment]
> I've been messing around for 20 minutes and getting a working ruby 2.7 environment set up has turned out to be harder than expected. You'll have to wait a bit more for me to review this.

I know the pain. Been there. Use docker, `docker run -v "$(pwd)"/:/app -p 4567:4567 -it ruby:2.7.8 /bin/bash` and install remaining things inside. Works fine.

> This was the latest version I could get to before bumping the bundler version

Could we maybe just bump everything? It's getting quite hard to test, deploy those old versions.

**Comment by frothedoatmilk** [comment]
Sure, that was my plan eventually, I just didn't want to change too many things in one PR.

**Comment by frothedoatmilk** [comment]
Fortunately no big changes! I had to put in webrick for a second but after updating middleman that issue went away so I dropped it from the Gemfile again. The depreciation warnings are not from us, all are middleman dependencies, so I didn't silence them.

**Comment by pigoz** [comment]
If the SassC issue is we need the following line, I think we can live with it.

```ruby
config[:sass_assets_paths] << Bootstrap.stylesheets_path
```

 This would probably not be needed with later versions of Bootstrap. BTW the website could probably be updated to not even need Bootstrap since CSS has come a long way since 2013.

**Comment by frothedoatmilk** [comment]
Would be interesting to see if bootstrap bumps will fix that. I'm very hesitant to do that in this PR though, SassC's 4 year EOL + two major bumps would definitely add to the diff.

File: README.md [hunk 1]
```diff
@@ -26,11 +26,10 @@ The website is created using a static website generator called
...
 copy of the website running on your local machine.
 
-- Optional: have ``rbenv`` or ``rvm`` installed. These tools will automatically
+- Strongly recommended: have ``rbenv`` or ``rvm`` installed. These tools will automatically
   select the correct version of Ruby based on the ``.ruby-version`` file.
-  Alternatively you will need to install Ruby 2.7 (**not** 3.x) manually.
-- ``gem install bundler:1.17.3``
-- ``bundle _1.17.3_ install``
-- ``rbenv rehash`` (only if using ``rbenv``)
+  Alternatively, just install the version in the [.ruby-version file](https://github.com/mpv-player/mpv.io/blob/master/.ruby-version) manually
```
Comment (sfan5) [inline README.md:31]: I tested this with 3.2.3 so I think we can relax this
```suggestion
  Alternatively you will need to install Ruby 3.x manually.
```

**Review by sfan5 — APPROVED** [review]
other than that: works. thanks for contributing.

**Comment by llyyr** [comment]
The manual no longer has the sidebar and options no longer have hyperlinks associated to them after this https://mpv.io/manual/master/

**Comment by pigoz** [comment]
I've pushed a fix but it looks like it's not being built.

**Comment by sfan5** [comment]
The workflow runs [here](https://github.com/mpv-player/mpv-player.github.io/actions) and every day at 8am. I'll trigger it manually now.

**Comment by pigoz** [comment]
Would it be welcome if I added Github Actions to automatically deploy on Cloudflare Pages so we can drop Github Pages? It even supports preview links on PRs with can be nice to test changes actually work.

**Pull request merged**